src/nvim/terminal.c

```diff
@@ -307,26 +309,68 @@ static int on_osc(int command, VTermStringFragment frag, void *user)
     return 1;
   }
 
-  StringBuilder request = KV_INITIAL_VALUE;
-  kv_printf(request, "\x1b]%d;", command);
-  kv_concat_len(request, frag.str, frag.len);
-  schedule_termrequest(term, request.items, request.size);
+  if (frag.initial) {
+    if (kv_size(term->termrequest_buffer) > 0) {
+      kv_drop(term->termrequest_buffer, kv_size(term->termrequest_buffer));
+    }
+    kv_printf(term->termrequest_buffer, "\x1b]%d;", command);
+  }
+  kv_concat_len(term->termrequest_buffer, frag.str, frag.len);
+  if (frag.final) {
+    schedule_termrequest(user, term->termrequest_buffer.items, term->termrequest_buffer.size);
```

Comment:
We should `xmemdupz` the termrequest buffer before passing to `schedule_termrequest`, otherwise there is a race condition (if `on_osc` is called again before `emit_termrequest` is called, you've just overwritten the buffer).

Reply:
Doesn't the `kv_init` call immediately after that line set the `item` pointer back to `NULL`?

Reply:
I looked at it again and `term->termrequest_buffer.items` gets set to `NULL`, completely disowning that pointer (it gets freed later in `emit_termrequest`. The next `on_*` call will then allocate a new one. I might be seriously misunderstanding something though.

Reply:
Good point, I'll change it.